lib/fluent/plugin/out_bigquery.rb

```diff
@@ -2,374 +2,285 @@
 
 require 'fluent/plugin/bigquery/version'
 
-require 'fluent/mixin/config_placeholders'
-require 'fluent/mixin/plaintextformatter'
-
 require 'fluent/plugin/bigquery/schema'
 require 'fluent/plugin/bigquery/writer'
 
-## TODO: load implementation
-# require 'fluent/plugin/bigquery/load_request_body_wrapper'
-
 module Fluent
-  ### TODO: error classes for each api error responses
-  # class BigQueryAPIError < StandardError
-  # end
-
-  class BigQueryOutput < TimeSlicedOutput
-    Fluent::Plugin.register_output('bigquery', self)
-
-    # https://developers.google.com/bigquery/browser-tool-quickstart
-    # https://developers.google.com/bigquery/bigquery-api-quickstart
-
-    ### default for insert
-    def configure_for_insert(conf)
-      raise ConfigError unless conf["method"] != "load"
-
-      conf["buffer_type"]                = "lightening"  unless conf["buffer_type"]
-      conf["flush_interval"]             = 0.25          unless conf["flush_interval"]
-      conf["try_flush_interval"]         = 0.05          unless conf["try_flush_interval"]
-      conf["buffer_chunk_limit"]         = 1 * 1024 ** 2 unless conf["buffer_chunk_limit"] # 1MB
-      conf["buffer_queue_limit"]         = 1024          unless conf["buffer_queue_limit"]
-      conf["buffer_chunk_records_limit"] = 500           unless conf["buffer_chunk_records_limit"]
-    end
-
-    ### default for loads
-    def configure_for_load(conf)
-      raise ConfigError unless conf["method"] == "load"
+  module Plugin
+    class BigQueryOutput < Output
+      Fluent::Plugin.register_output('bigquery', self)
+
+      helpers :inject, :event_emitter
+
+      # https://developers.google.com/bigquery/browser-tool-quickstart
+      # https://developers.google.com/bigquery/bigquery-api-quickstart
+
+      ### default for insert
+      def configure_for_insert(conf)
+        raise ConfigError unless conf["method"].nil? || conf["method"] == "insert"
+
+        buffer_config = conf.elements("buffer")[0]
+        return unless buffer_config
+        buffer_config["@type"]                       = "memory"      unless buffer_config["@type"]
+        buffer_config["flush_mode"]                  = :interval     unless buffer_config["flush_mode"]
+        buffer_config["flush_interval"]              = 0.25          unless buffer_config["flush_interval"]
+        buffer_config["flush_thread_interval"]       = 0.05          unless buffer_config["flush_thread_interval"]
+        buffer_config["flush_thread_burst_interval"] = 0.05          unless buffer_config["flush_thread_burst_interval"]
+        buffer_config["chunk_limit_size"]            = 1 * 1024 ** 2 unless buffer_config["chunk_limit_size"] # 1MB
+        buffer_config["total_limit_size"]            = 1 * 1024 ** 3 unless buffer_config["total_limit_size"] # 1GB
+        buffer_config["chunk_records_limit"]         = 500           unless buffer_config["chunk_records_limit"]
```

Comment:
It's better to have a comment for an URL of BQ's document to show why this default value is 500.